### PR TITLE
Determine if there is a winner or draw

### PR DESCRIPTION
**Pull request by kelseyroy**
Adds:
- Functionality to GamePlay that returns an end game state for 'win', 'draw', or null to continue playing
- Relevant tests for new GamePlay functionality
- Messages for an X win, O win and Draw situations
- Logic in TicTacToe to check for a win or draw after a player has marked the board
- Functionality in TicTacToe to display relevant end game messages

### CODE REVIEW

File: spec/game_play_spec.rb [hunk 1]
```diff
@@ -40,4 +39,102 @@
         expect(@game_play.select_spot(@board)).to eq(9)
     end
   end
+
```
Comment (t-keazirian) [inline spec/game_play_spec.rb:42]: Nice tests!!!  Very well organized, named clearly, and testing several cases - love to see it!

Comment (kelseyroy) [reply spec/game_play_spec.rb:42]: Thank you!!

File: lib/messages.rb [hunk 2]
```diff
@@ -4,7 +4,10 @@ class Message
       welcome: "Welcome to Tic Tac Toe!\n\n",
       instructions: "Tic Tac Toe is a two player game played\non a 3 x 3 grid. When prompted, each\nplayer will place their mark by entering\na number between 1-9 that corresponds with\nan available square on the grid.",
       get_spot: "Please select an available spot between 1-9:\n",
-      invalid_input: "That's not a valid spot.\n"
+      invalid_input: "That's not a valid spot.\n",
+      X_win: "Congratulations player X, you have won!\n",
+      O_win: "Congratulations player O, you have won!\n",
+      draw: "It's a draw!"
```
Comment (t-keazirian) [inline lib/messages.rb]: I got a `%` at the end of the line when I got a draw (but not when a player wins):
Screenshot 2023-04-17 at 7 00 48 PM

I see you have `\n` at the end of the other messages - maybe that would do the trick here too?

**Review by t-keazirian — APPROVED** [review]
Nice job on this one!  I played through it for all three scenarios (X winning, O winning, and a draw), as well as inputted incorrect inputs and all worked as expected!

Great job 🚢

File: lib/messages.rb (hunk 2 above)
Comment (kelseyroy) [reply lib/messages.rb]: Good catch! The next PR you see will make my messages more descriptive and tailored to specific players. But in my interest to get it working, I forgot to add the newline. I'll make the update now!

**Review by ghost — APPROVED** [review]
nice separation of classes and responsibilities

**Pull request merged**